plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,216 @@
+
+              submodules = helpers.defaultNullOpts.mkBool true "When false, git submodules will not be fetched. Defaults to true";
+
+              event =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+              cmd =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on command";
+
+              ft =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on filetype";
+
+              keys =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on key mapping";
+
+              module = helpers.defaultNullOpts.mkBool false "Do not automatically load this Lua module when it's required somewhere";
+
+              priority = helpers.defaultNullOpts.mkNum null "Only useful for start plugins (lazy=false) to force loading certain plugins first. Default priority is 50. It's recommended to set this to a high number for colorschemes.";
+
+              optional = helpers.defaultNullOpts.mkBool false "When a spec is tagged optional, it will only be included in the final spec, when the same plugin has been specified at least once somewhere else without optional. This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part of the user's plugins";
+
+              opts = helpers.mkNullOrOption (either attrs helpers.nixvimTypes.rawLua) "opts should be a table (will be merged with parent specs), return a table (replaces parent specs) or should change a table. The table will be passed to the Plugin.config() function. Setting this value will imply Plugin.config()";
+            };
+          });
+
+        listOfPlugins = types.listOf pluginType;
+      in
+        mkOption {
+          type = listOfPlugins;
+          default = [];
+          description = "List of plugins";
+        };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    extraPlugins = [(pkgs.vimPlugins.lazy-nvim.overrideAttrs (_: {pname = "lazy.nvim";}))];
```

Comment:
Why do we override the name?
If it's important there should be a comment

Reply:
Good question, will see what happens when I pull this.

Reply:
Removed the override, has no detrimental effect I can determine...